fits2caom2/src/main/java/ca/nrc/cadc/caom2/fits/wcs/Wcs.java

```diff
@@ -387,10 +387,20 @@ public static CoordFunction2D getCoordFunction2D(String utype, FitsMapping mappi
         Double cd12 = getDoubleValue(utype + ".cd12", mapping);
         Double cd21 = getDoubleValue(utype + ".cd21", mapping);
         Double cd22 = getDoubleValue(utype + ".cd22", mapping);
-        if (cd11 != null && cd22 != null && cd12 == null && cd21 == null)
+        if (cd11 != null || cd12 != null || cd21 != null || cd22 != null)
         {
-            cd12 = 0.0;
-            cd21 = 0.0;
+            if (cd11 == null) {
+                cd11 = 0.0;
+            }
+            if (cd12 == null) {
+                cd12 = 0.0;
+            }
+            if (cd21 == null) {
+                cd21 = 0.0;
+            }
+            if (cd22 == null) {
+                cd22 = 0.0;
+            }
```

Comment:
Please move the 
```return new CoordFunction2D(dimension, refCoord, cd11, cd12, cd21, cd22);```
inside this if (CD) block; it rightly puts everything inside a single "if (CD)" block.